docs/how-to/create-slice.rst

```diff
@@ -0,0 +1,120 @@
+
+Create your slice definition
+----------------------------
+
+You now have everything needed to actually define the OpenSSL slice that will install the content you are looking to have in the final ROCK.Since you are looking to install just the OpenSSL binaries from the ``openssl`` package, what about naming this new slice **bins**? Let's go for it:
+
+#. **What is the name of your slice definitions file?** It is a YAML file called ``openssl.yaml``
+#. **What package name should be defined inside this file?** The package name is ``openssl``
+#. **What is your slice name?** It should be called ``bins``
+#. **What contents do you need from the OpenSSL package?** Just the binaries - ``/usr/bin/c_rehash`` and ``/usr/bin/openssl``
+#. **Does your slice depend on any other package slice?** Yes, OpenSSL depends on ``libc6`` and ``libssl3``
+
+   * **Do these two packages have slice definitions files upstream?** Yes, there is already a slice definitions file for `libc6 <https://github.com/canonical/chisel-releases/blob/ubuntu-22.04/slices/libc6.yaml>`_ and another one for `libssl3 <https://github.com/canonical/chisel-releases/blob/ubuntu-22.04/slices/libssl3.yaml>`_ . If these dependencies were not present in the upstream Chisel release, you will also need to create their corresponding slice definitions
+   * **Which slices do you depend on then?** Since you only want the OpenSSL binaries, you might only need the libraries from ``libc6`` and ``libssl3``, as well as the configuration files from ``libc6`` and ``openssl`` itself.
+
+
+Create a new YAML file named ``openssl.yaml``, with the following content:
+
+.. code-block:: yaml
+
+    package: openssl
+    slices:
+        bins:
+            essential:
+              - libc6_libs
+              - libc6_config
+              - libssl3_libs
+              - openssl_config
+            contents:
+                /usr/bin/c_rehash:
+                /usr/bin/openssl:
+
+        config:
+            contents:
+                /etc/ssl/private/:
+                /etc/ssl/openssl.cnf:
+                /usr/lib/ssl/certs:
+                /usr/lib/ssl/openssl.cnf:
+                /usr/lib/ssl/private:
+
+Notice the unforeseen new slice ``config``. Because your OpenSSL binaries depend on the OpenSSL configuration files, and those were not yet present anywhere in the Chisel releases upstream, you also need to create that slice! You may also ask **"why not put those configuration files inside the ``bins`` slice"**? You could! But we recommend, as a best practice, to separate and group contents according to their nature, as you may tomorrow need to create a new slice definition that only needs the OpenSSL configurations and not the binaries.
```

Comment:
I think something that we need to address in documentation somehow in the future (not on this PR necessarily) is that the jump from "I just need the bins" to "I also need this bunch of config" feels a bit jarring. I know that someone (was it you @cjdcordeiro?) had to "suffer" when creating these slices, but on this how-to the new requirement feels a bit like it came out of nowhere.

In practice, that's probably going to be one of the harder parts when creating new slices, right? Is that something that would merit a separate how-to/troubleshooting/whatever section?

Reply:
This is really open to developer discretion. Like for DEBs, how do you say what pkgs your DEB depends on? First, the definition must exist, for one to be able to tell what does the slice depends on. In this case, that's the definition we are working on here...it does require some in-depth knowledge about the OpenSSL inner workings